easybuild/easyblocks/o/openfoam.py

```diff
@@ -361,6 +361,11 @@ def sanity_check_step(self):
                [os.path.join(toolsdir, "surface%s" % x) for x in ["Add", "Find", "Smooth"]] + \
                [os.path.join(toolsdir, x) for x in ['blockMesh', 'checkMesh', 'deformedGeom', 'engineSwirl',
                                                     'modifyMesh', 'refineMesh']]
+
+        # remove Boussinesq and sonic since for OpenFOAM >= 7, since those solvers are deprecated
+        if LooseVersion(self.version) >= LooseVersion('7'):
+            bins = [x for x in bins if "Boussinesq" not in x and "sonic" not in x]
```

Comment:
@mboisson Rather than adding them first and then removing then again, we should only add them for versions < `7`, something like:

```python
bins = [...
       [os.path.join(toolsdir, "buoyantSimpleFoam")] + \
       [os.path.join(toolsdir, "%sFoam" % x) for x in ["boundary", "engine"]] + \
       ...
]

# only include Boussinesq and sonic since for OpenFOAM < 7, since those solvers have been deprecated
if LooseVersion(self.version) < LooseVersion('7'):
    bins.extend([
        os.path.join(toolsdir, "buoyantBoussinesqSimpleFoam"),
        os.path.join(toolsdir, "sonicFoam")
    ])
```

Reply:
done.